### PR TITLE
Fix nested paragraphs

### PR DESCRIPTION
**Pull request by BarkingBad**

### CODE REVIEW

**Comment by BarkingBad** [comment]
Comparing to latest master on stdlib, git diff found exactly 6220 different lines, all have one paragraph less nested (if there were at least two). I assume it did not change any behaviour.

File: plugins/base/src/main/kotlin/parsers/MarkdownParser.kt [hunk 1]
```diff
@@ -465,46 +466,46 @@ class MarkdownParser(
             DocumentationNode(
                 (listOf(kDocTag) + getAllKDocTags(findParent(kDocTag))).map {
                     when (it.knownTag) {
-                        null -> if (it.name == null) Description(parseStringToDocNode(it.getContent())) else CustomTagWrapper(
-                            parseStringToDocNode(it.getContent()),
+                        null -> if (it.name == null) Description(parseStringToDocNode(it.getContent()) as RootDocTag) else CustomTagWrapper(
```
Comment (Kordyjan) [inline plugins/base/src/main/kotlin/parsers/MarkdownParser.kt:469]: Is it even possible for `parseStringToDocNode` to return something other than `RootDocTag`. If no move cast into that function. If this is possible consider slightly modifying tha api.

**Review by Kordyjan — CHANGES_REQUESTED** [review]

File: plugins/base/src/main/kotlin/parsers/MarkdownParser.kt (hunk 1 above)
Comment (BarkingBad) [reply plugins/base/src/main/kotlin/parsers/MarkdownParser.kt:469]: Yeah, I shouldn't have left it that way, will refactor soon

**Comment by sellophane** [comment]
1) Please squash commits before merge
2) `Fix tests` breaks compilation

File: plugins/base/src/main/kotlin/transformers/pages/comments/DocTagToContentConverter.kt [hunk 2]
```diff
@@ -175,7 +176,18 @@ object DocTagToContentConverter : CommentsToContentConverter {
...
+            else -> when {
+                docTag is CustomDocTag && docTag.name == MarkdownElementTypes.MARKDOWN_FILE.name -> if (docTag.children.size > 1)
+                    listOf(ContentGroup(
+                        buildChildren(docTag),
+                        dci,
+                        sourceSets.toDisplaySourceSets(),
+                        styles,
+                        extra = extra
+                    )) else buildChildren(docTag)
+                else -> buildChildren(docTag)
```
Comment (kamildoleglo) [inline plugins/base/src/main/kotlin/transformers/pages/comments/DocTagToContentConverter.kt:189]: Extract to separate method

File: runners/gradle-plugin/src/test/kotlin/org/jetbrains/dokka/gradle/AndroidAutoConfigurationTest.kt [hunk 3]
```diff
@@ -61,6 +61,7 @@ class AndroidAutoConfigurationTest {
         }
     }
 
+    @Ignore
```
Comment (kamildoleglo) [inline runners/gradle-plugin/src/test/kotlin/org/jetbrains/dokka/gradle/AndroidAutoConfigurationTest.kt:64]: Please remove

**Review by kamildoleglo — CHANGES_REQUESTED** [review]

**Pull request merged**